### PR TITLE
Cleanup pyscript web elements

### PR DESCRIPTION
**Pull request by fpliger**
## Checklist

-   [ ] All tests pass locally
-   [ ] I have updated `CHANGELOG.md`
-   [ ] I have created documentation for this(if applicable)

### CODE REVIEW

**Comment by WebReflection** [comment]
@fpliger there is a merge conflict but the other one looked OK to me after you rlatest cleanup so I've merged it but I wouldn't know how to resolve those conflicts ... any clue?

**Comment by WebReflection** [comment]
@fpliger I thought you made it back to ready for review (but it was already approved) to have it in ... my bad, I should've waited or asked more ... we just released though, we have time to make changes and fixes ... arguably we could also reverse that merge, that's what git is good at :wink: 

if you have everything under control though, I let you fix the current state, thanks

**Comment by fpliger** [comment]
ehehe, no worries at all. My bad really.. we were rushing yesterday evening because it was already late so I ended up forgetting both to mention that in the PR and not opening the other PR towards this one. I'm working on it today with @mchilvers and I hope we can get `main` to a stable state today or tomorrow the latest. 👍  (== before we release again)

**Comment by WebReflection** [comment]
@fpliger @mchilvers @ntoll latest code that made the release possible without having a broken PyWeb in it, required some minor change that is going to inevitably conflict but fear not, it should be fairly straight forward for you to be again able to work on this.

The [event_handling.py](https://github.com/pyscript/pyscript/pull/2105/files#diff-2cd036a46f612058d14a1afd99508374f4c220e41f6e1d400a28cbf03adef914) should be updated to keep your latest changes and so should the [test/pydom.py](https://github.com/pyscript/pyscript/pull/2105/files#diff-3ba63a57ec7b96761775e974dbbad877955ebbce2d0d3a7193ca03d8b18d96fc) file and [tests/test_dom.py](https://github.com/pyscript/pyscript/pull/2105/files#diff-63d754b1fa6dc76984dd157f58528c90c3a03e5c7c7790c8e7705c5c90fd9eb8) plus [test/integration/test_pyweb.py](https://github.com/pyscript/pyscript/pull/2105/files#diff-8e40d72e376b00cd2fbfb063c5a7a22dd8aa5098ac8ee6e6a8973316ac45b184)

Everything else can stay the way it is but once this moves forward, the `pyweb` folder in `stdlib` should be removed too. Right now we have both old pyweb and current one (which is not finished yet) to avoid breaking this MR but of course that adds to the code landed as Python dependency for our module.

The TL;DR is that you should rebase and keep all your changes on top, or you can just copy those files somewhere, rebase bringing in the old tests and replace with a different commit to be back on track.

Happy to help further, if needed, but beside tests and that `event_handling.py` the rest is basically simply split in folders and well organized, yet tests don't know that so that's the only little friction.

On the other side, because most code is repeated, after gzip/brotli the change is not super relevant for our users, still shipping code that is not necessary isn't ideal in general.

**Review by ntoll — COMMENTED** [review]
@WebReflection / @fpliger please have a quick look over this. @mchilvers and I paired on this yesterday and we both feel it's in a ready-to-merge state with a few refinements for the future.

Top of my mind for future refinement is a sort of "minification" strategy in our build step:

* Python indentation from four spaces to one space.
* Remove all `# comments`
* De-`black` multiline parenthesis statements. E.g.
```python
x = ("foo", "bar", "baz")  # this
x = (  # Not this
  "foo",
  "bar",
  "baz",
)
```

A good starting place for this might be a tool like [pyminifier](https://pyminifier3.readthedocs.io/en/latest/) which does most of what we want in a vaguely configurable manner. :+1:

I'm going to update the docs in line with the changes in this PR.

File: pyscript.core/src/stdlib/pyscript/web/elements.py [hunk 1]
```diff
@@ -192,187 +191,178 @@ def content(self, value):
...
 
     @property
-    def id(self):
-        return self._js.id
-
-    @id.setter
-    def id(self, value):
-        self._js.id = value
+    def children(self):
+        return [element_from_dom(el) for el in self._dom_element.children]
```
Comment (ntoll) [inline pyscript.core/src/stdlib/pyscript/web/elements.py:195]: Should this not be an `ElementCollection` instance..?

**Review by mchilvers — APPROVED** [review]
